extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -359,4 +360,21 @@ private boolean hasEntities(JpaModelBuildItem jpaModel) {
         return !jpaModel.getEntityClassNames().isEmpty();
     }
 
+    /**
+     * This registers a part of JTA Context Propagation to be unremovable, because our quarkus-hibernate-orm imports excludes
+     * quarkus-narayana, which would mark it as unremovable. This means that if by accident or other, the
```

Comment:
"because our quarkus-hibernate-orm imports excludes quarkus-narayana, which would mark it as unremovable."

I don't get what you're suggesting here. Having an exclusion is marking stuff as unremovable?

Reply:
No, it's quarkus-narayana's job to mark the bean as unremovable. If we don't import it, it's removed and then SR-CP-JTA fails.

Reply:
I still don't get it.

You're saying there is a dependnecy of `smallrye-context-propagation-jta` to `quarkus-narayana` .. ok that's reasonable so far.

But then why apply a workaround in the Hibernate Reactive extension? This code is totally agnostic to JTA.

If `smallrye-context-propagation-jta` requires this bean to be set as unremoveable, shouldn't you add this build step in the extension of smallrye-context-propagation-jta ?

Reply:
There's no such extension. Well, not for Quarkus. `smallrye-context-propagation-jta` on its own would work, but not due to `hibernate-reactive` excluding `quarkus-narayana` from its import. So, because `hibernate-reactive` is the most likely module that users will use when they get this error, and it's due to its exclusion, it sounds reasonable to add this workaround there, especially since it doesn't introduce any dependency and shouldn't cost anything, no?

Reply:
> There's no such extension. 

oh, ok.

But you do know that exclusions aren't going to affect other modules which depend on it right?

Reply:
If other modules depend on `quarkus-hibernate-orm` then it will work fine anyway.

Reply:
> If other modules depend on `quarkus-hibernate-orm` then it will work fine anyway.

I mean of course modules which don't depend on `quarkus-hibernate-orm` ,don't depend on ``quarkus-hibernate-orm-reactive` or any other extension which brings in JTA.

What happens to the pour souls who have context propagation but don't have any of these modules?

Reply:
They won't have any issue if they don't have extensions which trigger an import of `smallrye-context-propagation-jta`.

Reply:
so why is this import triggered ? How does that work?

I thought we didn't do conditional dependencies yet, did I miss something?